Required wp_slash() if WordPress importer is NOT exist.

maheshwaghmare opened the pull request:
#12186  PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

Slashes are required if WordPress importer is NOT exist.

*

## Description

Regarding the issue https://github.com/elementor/elementor/issues/11466 the fix is valid.

But, There is one condition missing which is below:

```
if( empty( $wp_importer ) ) {
	return true;
}
```

If there is WordPress importer exist then we can skip the **wp_slash()**.

But, If it is NOT exist then we need **wp_slash()** for the `_elementor_data`.

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [] I have tested this code to the best of my abilities
- [] I have added unittests to verify the code works as intended
- [] Docs have been added / updated (for bug fixes / features)

Fixes #11466

maheshwaghmare committed:
Required wp_slash() if WordPress importer is NOT exist.

Regarding the issue https://github.com/elementor/elementor/issues/11466 the fix is valid.

But, There is one condition missing which is below:

```
if( empty( $wp_importer ) ) {
	return true;
}
```

If there is WordPress importer exist then we can skip the **wp_slash()**.

But, If it is NOT exist then we need **wp_slash()** for the `_elementor_data`.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/elementor/elementor?pullRequest=12277) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/elementor/elementor?pullRequest=12277) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/elementor/elementor?pullRequest=12277) it.</sub>

udidol commented:
Hi @maheshwaghmare,

This method is only run when the `wp_import_post_meta` or the `wxr_importer.pre_process.post_meta` hooks are called. These hooks are called by the WP Importer (or extensions of it). 

The compatibility functions that use the `is_wp_importer_before_0_7()` method are there to handle compatibility issues with WP Importer, so if WP importer does not exist, they should not intervene. That is why we CANNOT return `true` if WP Importer is not found.

If you are sending this PR because you came across a problem with importing Elementor content, please feel free to open a ticket in the Issues tab with a detailed explanation of your problem.

Thank you for your interest in contributing!

maheshwaghmare commented:
Actually, In "[Starter Templates](https://wordpress.org/plugins/astra-sites/)" plugin we have used the extended version [WXR_Importer](https://github.com/brainstormforce/astra-sites/blob/master/inc/importers/wxr-importer/class-wxr-importer.php#L13) of class WP_Importer. 

The WXR_Importer has the same filters which are used by default WordPress Importer.

Now, In Elementor version 3.0.0 and above the filters  `wp_import_post_meta` and `wxr_importer.pre_process.post_meta` add the slashes depending on the WordPress Importer plugin.

We have also added the site import support through WP CLI. While importing the site with the WP CLI command, we face the same issue. Because In Elementor the the filter `wp_import_post_meta` and `wxr_importer.pre_process.post_meta` works for only `is_admin() || defined( 'WP_LOAD_IMPORTERS' )`. So, For WP CLI site import, we have added the same filter into our compatibility file.

Now, To add compatibility for Elementor version 3.0.0 and above, we have added the quick fix in [/astra-sites/../class-astra-sites-compatibility-elementor.php#L58](https://github.com/brainstormforce/astra-sites/blob/next-release/inc/classes/compatibility/elementor/class-astra-sites-compatibility-elementor.php#L58).

For now, We have added the fix for the Elementor version 3.0.0 and above, and it works as expected. But, If some other developers also use the same way, then this may create conflict.

udidol commented:
I see.
Since the manipulation we do using those hooks is only done on Elementor content, then any external plugins that deal with Elementor content would indeed have to make sure they are compatible with changes we make to Elementor, including things like imports.

If you are not signed up to our mailing list, I recommend it - at [](https://developers.elementor.com/). Any major changes made are published here and any important announcements are sent to our mailing list.
